### ISSUE TITLE
use tcmalloc explicitly without overriding system malloc/free

### ISSUE THREAD

**Issue by sillykelvin** [gperftools/gperftools#1067]
In our application, we want `malloc/free` calls using system ones(it's glibc here), and when we want to use tcmalloc, we can call `tc_malloc/tc_free` explicitly. However, after we linked to tcmalloc with `-ltcmalloc_minimal`, it seems all `malloc/free` have been replaced.

I searched with google but nothing useful found, is there any way to achieve this?

EDIT:
We are using a Linux box with kernel version 3.10, glibc version 2.17, gcc version 4.8.5, and tcmalloc is compiled with `--enable-minimal` flag, and is statically linked into our program.

**Comment by kelvinh** [comment]
I had a look at the code, and found the `__attribute__(alias(...))` thing, and tried commenting out the following lines in `tcmalloc.cc`, and it now does not implicitly replace system `malloc/free` anymore:

```
// ...

using STL_NAMESPACE::vector;

// #include "libc_override.h"  // <--- commented out this line

using tcmalloc::AlignmentForSize;

// ...

TCMallocGuard::TCMallocGuard() {
  if (tcmallocguard_refcount++ == 0) {
    // ReplaceSystemAlloc();    // defined in libc_override_*.h // <--- commented out this line
    tc_free(tc_malloc(1));

// ...
```

However, I think this is just a **DIRTY HACK**, I still want to know if there is a more elegant solution.

**Comment by alk** [comment]
Yes, we don't officially support this. But because our windows bits already add necessary ifdef-s, I think it should be easy to provide compile-time option to disable aliasing to operator new/delete & malloc/free/etc. So feel free to just propose patch for that.

**Comment by anqur** [comment]
I used `env CPPFLAGS="-DTCMALLOC_SKIP_OVERRIDE" ./configure --enable-frame-pointers --disable-shared --enable-static --with-pic` for building my Julia compiler and it worked, and `--enable-frame-pointers` here is necessary for JIT systems otherwise `backtrace()` will deadlock.
